File: 07_Trees/10_Count_Good_Nodes_in_Binary_Tree.py
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        self.ans = 0

        def dfs(root, maxVal):
            if not root:
                return
            if root.val >= maxVal:
                maxVal = root.val
                self.ans += 1
            
            dfs(root.left, maxVal)
            dfs(root.right, maxVal)
        
        dfs(root, root.val)
        return self.ans

File: 07_Trees/10_Count_Good_Nodes_in_Binary_Tree.cpp
class Solution
{
public:
    int ans;

    void dfs(TreeNode *root, int maxVal)
    {
        if (!root)
            return;
        if (root->val >= maxVal)
        {
            ans++;
            maxVal = root->val;
        }
        dfs(root->left, maxVal);
        dfs(root->right, maxVal);
    }

    int goodNodes(TreeNode *root)
    {
        dfs(root, root->val);
        return ans;
    }
};